src/components/RadioForm/SeparationNP.tsx

```diff
@@ -0,0 +1,176 @@
+  errors: any
+  register: (name: string) => any
+  setValue: (name: string, value: any) => void
+  clearErrors: (name: string) => void
+}
+
+const getCityRef = (value: string, novaPoshtaCity: any[]) => {
+  const [descriptionPart, areaPart] = value.split(' (')
+  const mainDescription = descriptionPart.replace(/^(м\.|с\.)\s/, '')
+  const area = areaPart.replace(' обл.', '').replace(')', '')
+
+  const selectedCity = novaPoshtaCity[0]?.Addresses.find(
+    (address: any) =>
+      address.MainDescription === mainDescription && address.Area === area
+  )
+
+  return selectedCity ? selectedCity.Ref : null
+}
+
+const getDefaultCityRef = (value: string, cityDefault: any[]) => {
+  const defaultCity = cityDefault.find((city) => city.name === value)
+  return defaultCity ? defaultCity.ref : null
+}
+
+export const SeparationNP: FC<PropsType> = ({
+  selectedValue,
+  errors,
+  register,
+  setValue,
+  clearErrors
+}) => {
+  const {
+    valInputCity,
+    setValInputCity,
+    loading: cityLoading,
+    options,
+    novaPoshtaCity,
+    cityDefault
+  } = useNovaPoshtaCity()
+  const [settleRef, setSettleRef] = useState<string | null>(null)
+  const [warehouses, setWarehouses] = useState<any[]>([])
+  const [valInputWarehouse, setValInputWarehouse] = useState<string>('')
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const disabledBtn = () => {
+    if (!valInputCity) {
+      return true
+    }
+  }
+
+  useEffect(() => {
+    if (!valInputWarehouse) {
+      setWarehouses([])
+    }
+  }, [valInputWarehouse])
+
+  const fetchWarehouses = async (val: string) => {
+    setLoading(true)
+    try {
+      if (settleRef) {
+        const response = await axiosInstance.get(
+          '/api/nova_poshta/warehouses/branches/',
+          {
+            params: {
+              settle_ref: settleRef,
+              search_term: val
+            }
+          }
+        )
+        console.log('✌️response --->', response)
+        setWarehouses(response.data)
+      }
+    } catch (error) {
+      console.log('✌️error --->', error)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  useEffect(() => {
+    if (valInputWarehouse) {
+      const timer = setTimeout(() => {
+        fetchWarehouses(valInputWarehouse)
+      }, 1000)
+
+      return () => clearTimeout(timer)
+    }
+  }, [valInputWarehouse])
+
+  const MAX_LENGTH = 50
+
+  const optionsDataCity = useMemo(() => {
+    return warehouses.map(
+      (warehouse) =>
+        warehouse.address_warehouse
+          .replace(/\(до 30 кг на одне місце\)/g, '')
+          .replace(/\(до 30 кг\)/g, '')
+          .replace(/\(до 10 кг\)/g, '')
+          .replace(/\(до 5 кг\)/g, '')
+          .replace(/\(до 200 кг\)/g, '')
+          .replace(/\(до 1100 кг \)/g, '')
+          .replace(/\n/g, '')
+          .replace(/№(\d+)\s*:/g, '№$1:')
+          .trim()
+          .slice(0, MAX_LENGTH) +
+        (warehouse.address_warehouse.length > MAX_LENGTH ? '...' : '')
+    )
```

Comment:
move this to `utils` as a helper functions

Reply:
done